File: leetcode/1201-1500/1251-1300/1281.Subtract the Product and Sum of Digits of an Integer.cc
// https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/

class Solution {
public:
    int subtractProductAndSum(int n) {
        int product = 1;
        int sum = 0;
        int div = 10;
        
        while (n != 0) {
            int num = n % div;
            product *= num;
            sum += num;
            n /= div;
        }
        
        return product - sum;
    }
};

File: leetcode/1201-1500/1251-1300/1281.Subtract the Product and Sum of Digits of an Integer.rs
// https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer

impl Solution {
    pub fn subtract_product_and_sum(n: i32) -> i32 {
        let mut sum = 0;
        let mut prod = 1;
        let mut div = 1;
        while n / div > 0 {
            sum += (n % (div * 10) / div);
            prod *= (n % (div * 10) / div);
            div *= 10;
        }

        prod - sum
    }
}